pkg/util/parsers/parsers_test.go

```diff
@@ -38,10 +39,16 @@ func TestParseImageName(t *testing.T) {
 		{Input: "url:5000/repo", Repo: "url:5000/repo", Tag: "latest"},
 		{Input: "url:5000/repo:tag", Repo: "url:5000/repo", Tag: "tag"},
 		{Input: "url:5000/repo@sha256:e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855", Repo: "url:5000/repo", Digest: "sha256:e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855"},
+		{Input: "ROOT", expectedError: true},
+		{Input: "http://root", expectedError: true},
+		{Input: "e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855", expectedError: true},
 	}
 	for _, testCase := range testCases {
 		repo, tag, digest, err := ParseImageName(testCase.Input)
-		if err != nil {
+		if testCase.expectedError && err != nil {
+		} else if testCase.expectedError && err == nil {
+			t.Errorf("ParseImageName(%s) expects error but did not get one", testCase.Input)
+		} else if err != nil {
```

Comment:
Maybe it's just me but this empty if branch looks a bit odd

```suggestion
		if testCase.expectedError && err == nil {
			t.Errorf("ParseImageName(%s) expects error but did not get one", testCase.Input)
		} else if !testCase.expectedError && err != nil {
```

also using swtich-case might make it a bit more readable

```suggestion
		switch {
		case testCase.expectedError && err == nil:
			t.Errorf("ParseImageName(%s) expects error but did not get one", testCase.Input)
		case !testCase.expectedError && err != nil:
```

Reply:
Good call.  Will push a change.

Reply:
Used switches and add error checks.